File: CS-Examples/10_Conversion/HtmlToImage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using System.Drawing.Imaging;

namespace HtmlToImage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create Word document.
			Document document = new Document();

			//Load the file from disk.
			document.LoadFromFile(@"..\..\..\..\..\..\Data\Template_HtmlFile1.html", FileFormat.Html, XHTMLValidationType.None);

			string result = "Result-HtmlToImage.png";

            //Save to image. You can convert HTML to BMP, JPEG, PNG, GIF, Tiff£¬etc.
            Image image = document.SaveToImages(0, ImageType.Bitmap);
			image.Save(result, ImageFormat.Png);

            //////////////////Use the following code for netstandard dlls/////////////////////////
            /*
            SkiaSharp.SKImage images = document.SaveToImages(0, ImageType.Bitmap);
            FileStream fileStream = new FileStream(outputFile, FileMode.Create, FileAccess.Write);
            images.Encode(SkiaSharp.SKEncodedImageFormat.Png, 100).SaveTo(fileStream);
            fileStream.Flush();
            */

            //////////////////Use the following code for WPF dlls/////////////////////////
            /*
            BitmapSource[] images = document.SaveToImages(Spire.Doc.Documents.ImageType.Bitmap);
            PngBitmapEncoder pE = new PngBitmapEncoder();
            pE.Frames.Add(BitmapFrame.Create(images[0]));
            string outputfile = String.Format(outputFile, ImageFormat.Png);
            using (Stream stream = File.Create(outputfile))
            {
                pE.Save(stream);
            }
            */

            //Dispose the document
            document.Dispose();

            //Launch the image.
            WordDocViewer(result);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/10_Conversion/HtmlToImage.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports System.Drawing.Imaging
Imports System.IO

Namespace HtmlToImage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Create Word document.
			Dim document As New Document()

			'Load the file from disk.
			document.LoadFromFile("..\..\..\..\..\..\Data\Template_HtmlFile1.html", FileFormat.Html, XHTMLValidationType.None)

			Dim result As String = "Result-HtmlToImage.png"

			'Save to image
			Dim image As Image = document.SaveToImages(0, ImageType.Bitmap)
			image.Save(result, ImageFormat.Png)
			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			'Dim images As SkiaSharp.SKImage = document.SaveToImages(0, ImageType.Bitmap)
			'Using fileStream As New FileStream(outputFile, FileMode.Create, FileAccess.Write)
			'	images.Encode(SkiaSharp.SKEncodedImageFormat.Png, 100).SaveTo(fileStream)
			'	fileStream.Flush()
			'End Using
			' =============================================================================

			' =============================================================================
			' Use the following code for WPF dlls
			' =============================================================================
			'Dim images() As BitmapSource = document.SaveToImages(Spire.Doc.Documents.ImageType.Bitmap)
			'Dim pE As New PngBitmapEncoder()
			'pE.Frames.Add(BitmapFrame.Create(images(0)))
			'Dim outputfile As String = String.Format(outputfile, ImageFormat.Png)
			'Using stream As Stream = File.Create(outputfile)
			'	pE.Save(stream)
			'End Using
			' =============================================================================


			'Dispose the document
			document.Dispose()

			'Launch the image.
			WordDocViewer(result)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
